airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/write/Batch.kt

```diff
@@ -0,0 +1,14 @@
+package io.airbyte.cdk.write
+
+/**
+ * Represents an accumulated batch of records in some stage of processing.
+ *
+ * Emitted by the record accumulator per batch accumulated. Handled by
+ * the batch processor, which may advanced the state and yield a new batch.
+ *
+ * If persisted is true, the framework will close state for the accumulated messages.
+ */
+open class Batch(
+    val name: String,
+    val persisted: Boolean = false
+)
```

Comment:
nit: If streams have names+namespaces, and batch names need to be unique, maybe batches have a namespace too?

Reply:
Or, a UUID id?

Reply:
In the current model, batch names don't need to be unique, they just convey whatever the user wants them to.

However, given ed's comment [here](https://github.com/airbytehq/airbyte/pull/44122#discussion_r1718895878), we might want to think of them as task-ids and let the user set concurrency limits if desired.